sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/HashUtils.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf
+package crypto
+
+import com.daml.scalautil.Statement.discard
+import com.digitalasset.daml.lf.crypto.HashUtils.HashTracer.StringHashTracer.{
+  encodedValueWrapper,
+  encodingRegex,
+  inlineContext,
+  inlineValue,
+}
+import com.digitalasset.daml.lf.data.Bytes
+
+import java.nio.ByteBuffer
+
+object HashUtils {
+
+  private val lineSeparator = System.getProperty("line.separator")
+  private[crypto] def formatByteToHexString(byte: Byte): String = String.format("%02X", byte)
+
+  /** Interface to provide observability in the encoding algorithm.
```

Comment:
I think this was a good idea to add, helps a lot reading the expected hashing in your tests